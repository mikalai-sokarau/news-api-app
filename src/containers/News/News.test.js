import React from 'react';
import { shallow } from 'enzyme';
import { News } from './index';

describe('<News />', () => {
  const fakeNews = [{ url: 'url1' }, { url: 'url2' }, { url: 'url3' }];

  test('should be rendered', () => {
    expect(
      shallow(
        <News
          news={fakeNews}
          addNewsToFavorite={() => {}}
          favoriteNewsKeys={[]}
          getNewsFrom={() => {}}
        />
      )
    ).toMatchSnapshot();
  });

  test('callback should be executed when `componentDidMount` fires', () => {
    const callback = jest.fn();

    shallow(
      <News
        news={fakeNews}
        addNewsToFavorite={() => {}}
        favoriteNewsKeys={[]}
        getNewsFrom={callback}
      />
    );
    expect(callback).toHaveBeenCalled();
  });

  test('news array should be rendered as SingleNews components', () => {
    expect(shallow(
      <News
        news={fakeNews}
        addNewsToFavorite={() => {}}
        favoriteNewsKeys={[]}
        getNewsFrom={() => {}}
      />
    ).find('SingleNews')).toHaveLength(3);
  });
});
