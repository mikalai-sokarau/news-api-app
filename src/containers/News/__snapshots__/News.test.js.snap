// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<News /> should be rendered 1`] = `
<div
  className="col s8"
>
  <Error>
    <div>
      <SingleNews
        addNewsToFavorite={[Function]}
        item={
          Object {
            "url": "url1",
          }
        }
        key="url1"
      />
      <SingleNews
        addNewsToFavorite={[Function]}
        item={
          Object {
            "url": "url2",
          }
        }
        key="url2"
      />
      <SingleNews
        addNewsToFavorite={[Function]}
        item={
          Object {
            "url": "url3",
          }
        }
        key="url3"
      />
    </div>
  </Error>
</div>
`;
